wagtail_footnotes/blocks.py

```diff
@@ -37,17 +40,28 @@ def replace_footnote_tags(self, value, html, context=None):
         page = new_context["page"]
         if not hasattr(page, "footnotes_list"):
             page.footnotes_list = []
```

Comment:
I think we can do away with `footnotes_list` in favour of the new `footnotes_references`.

Reply:
These are two different things. `footnotes_list` is the list of the actual Footnote objects to be rendered at the bottom of the rendered page, and `footnotes_references` is a dict that maps a Footnote's UUID to a list of all of the references found in the rich text blocks, so that jump links back to each individual reference can be rendered. 

I didn't have any better idea of how to make the Footnote objects aware of the one or more references contained in the rich text blocks. This is why I made the goofy template tag so I could at least do dict lookups using the UUID in the template to fetch the references.

I'm working on getting this rebased on main now that the custom template PR was merged.

Reply:
Right, that makes sense. I think some additional comments would be helpful then, and perhaps some type hinting, then there will be no ambiguity. Thank you